### DISCUSSION TITLE
SwitchAction setzen aus Datenpunkt der true/false liefert

### DISCUSSION THREAD

**Discussion by robert-ke** [Zefau/ioBroker.jarvis#756] [Ideas / Feature Requests]
Hallo zusammen,

ich hänge im Moment fest - habe einen Datenpunkt der true/false liefert und über eine Anzeigevariante den Text angepasst:

![image](https://user-images.githubusercontent.com/74955542/118355704-b2c32f00-b571-11eb-9463-d8e37eef6e17.png)

Leider geht der Button im Layout nicht auf AN:
![image](https://user-images.githubusercontent.com/74955542/118355714-cc647680-b571-11eb-9975-850e2f23367f.png)

Der Text ändert sich schon sauber mit, wenn der Datenpunkt geändert wird von läuft auf aus.
Wo vergesse oder übersehe ich was?

Vielen Dank für eure Hilfe & Gruß,
Robert

**Comment by Zefau** [comment]
Die Datenpunkt Eigenschaften sehen falsch aus. Warum ist das gesetzt? Es muss nur gesetzt werden, wenn der Datenpunkt nicht true / false für an / aus nutzt. Geht es, wenn du das raus nimmst?

**Reply by robert-ke** [reply]
Vielen Dank für den Tipp :-)

Eben getestet - ja, in der Tat geht der Button dann mit:
![image](https://user-images.githubusercontent.com/74955542/118356174-4138b000-b574-11eb-89e1-9e98d126f5d2.png)

Krieg ich das trotzdem unter einen Hut?
Ich brauche die Möglichkeit, definierte Texte für ON Action und OFF Action in den Trigger-Datenpunkt zu schreiben. Meine Datenpunkte sind voneinander entkoppelt - heißt, ich habe einen Datenpunkt für die Anzeige der liefert den Status und muss in einen anderen einen Text für die Action reinschreiben - das geht dann zu einer SPS... in diesem Beispiel wäre das wie folgt:

Switch Action auf Ein -> TVTerrasseStromEin
Switch Action auf Aus -> TVTerrasseStromAus

beides muss in einen Trigger-Datenpunkt, ein Script schickt die Befehle an die SPS weiter....

**Comment by Zefau** [comment]
Das geht momentan nicht, aber interessanter Use Case. Werde ich bei v3 berücksichtigen.

**Reply by robert-ke** [reply]
OK, super - wäre ein tolles Feature.
Bislang hab ich nur den ButtonAction verwendet - und damit ein "umschalten" realisiert - sprich, ich schicke auf die SPS ein TVTerrasseStromUmschalten hin - aber dann muss ich das in der SPS entsprechend abhandeln und schöner wäre einfach der Switch :-)

Danke & ein schönes Wochenende

**Comment by robert-ke** [comment]
Hallo @Zefau ,

gibts dazu inzwischen schon eine Entkopplung? Ansonsten hänge ich ein paar Scripte dahinter.... habe es bis dato nur noch offen gelassen.

**Reply by Zefau** [reply]
Noch nicht, aber kommt mit v3, siehe https://github.com/Zefau/ioBroker.jarvis/issues/757

**Reply by Zefau** [reply]
@robert-ke ist mit `v3.0.0-alpha.71` nun implementiert, siehe Feature Request
